compiler/parser.jou

```
@@ -298,13 +334,42 @@ class Parser:
                 self->tokens++
 
                 result = AstType{
-                    location = location,
                     kind = AstTypeKind.Array,
                     array = AstArrayType{
                         member_type = p,
                         length = len_expression,
                     }
                 }
+            else:
+                # Generic, e.g. List[int]
+                free(p)
+
+                nparams = 0
+                params: AstType* = NULL
+                while not self->tokens->is_operator("]"):
+                    params = realloc(params, sizeof(params[0]) * (nparams + 1))
+                    params[nparams++] = self->parse_type()
+                    if not self->tokens->is_operator(","):
+                        break
+                    self->tokens++
+
+                if nparams == 0:
+                    fail(self->tokens->location, "empty [] is not valid syntax when specifying a type")
+
```

Let's just let it fail later

```suggestion
```